src/gui/folder.cpp

```diff
@@ -314,6 +314,62 @@ void Folder::etagRetreivedFromSyncEngine(const QString &etag)
     _lastEtag = etag;
 }
 
+void Folder::slotAccountCapabilitiesChanged()
+{
+    const Capabilities& capabilities = _accountState->account()->capabilities();
+    QString wsUrl = capabilities.getWebSocketUrl();
+    if (wsUrl.isEmpty()) {
+        return;
+    }
+
+    if (_webSocket) {
+        return;
+    }
+    //initialize wamp connection
+    _webSocket.reset(new QTcpSocket());
+    QUrl webSocketUrl(wsUrl);
+    QObject::connect(_webSocket.data(), &QTcpSocket::connected, [&]() {
+
+        QString sessionName("sessionName");
+        _session.reset(new QWamp::Session(sessionName, *_webSocket.data(), QWamp::Session::MessageFormat::Msgpack, true));
+
+        QObject::connect(_session.data(), &QWamp::Session::joined, [&](qint64 s) {
+            qDebug() << "Session joined to realm1 with session ID " << s;
+
+            // TODO: add current userid
+            _session->subscribe("files.root-etag-changed.admin", [&](const QVariantList& args, const QVariantMap& options) {
+                qDebug() << "Event received";
+                QString notificationEtag = args.at(0).toString();
```

Comment:
`QList::at`  crashes if the list don't have enough argument, which can happen, i guess, if the server is sending garbage.
So you can use `value()` which will return an empty variant if out of bounds.  or you should check if there is enough element in the list before.